jit/jit.cpp

```diff
@@ -0,0 +1,467 @@
+#include "jit.h"
+
+#include <sys/mman.h>
+#include <boost/noncopyable.hpp>
+#include <llvm/Analysis/TargetTransformInfo.h>
+#include <llvm/IR/BasicBlock.h>
+#include <llvm/IR/DataLayout.h>
+#include <llvm/IR/DerivedTypes.h>
+#include <llvm/IR/Function.h>
+#include <llvm/IR/IRBuilder.h>
+#include <llvm/IR/Mangler.h>
+#include <llvm/IR/Type.h>
+#include <llvm/IR/LegacyPassManager.h>
+#include <llvm/ExecutionEngine/JITSymbol.h>
+#include <llvm/ExecutionEngine/SectionMemoryManager.h>
+#include <llvm/ExecutionEngine/JITEventListener.h>
+#include <llvm/MC/SubtargetFeature.h>
+#include <llvm/MC/TargetRegistry.h>
+#include <llvm/Support/DynamicLibrary.h>
+#include <llvm/Support/Host.h>
+#include <llvm/Support/TargetSelect.h>
+#include <llvm/Transforms/IPO/PassManagerBuilder.h>
+#include <llvm/Support/SmallVectorMemoryBuffer.h>
+#include <llvm/Passes/PassBuilder.h>
+
+#include <llvm/ExecutionEngine/RTDyldMemoryManager.h>
+#include <llvm/ADT/None.h>
+#include <llvm/ExecutionEngine/MCJIT.h>
+
+#include <boost/noncopyable.hpp>
+namespace msc_jit
+{
+// Arena Memory pool for JIT
+class Arena : private boost::noncopyable
+{
+ public:
+  Arena() : page_size(8)
+  {
+  }
```

Comment:
We could live with an implicit default constructor if the initializer expression for `page_size` was revised to this value.